File: frontend/inundago-frontend/src/app/core/user/user.service.ts
import { Injectable } from "@angular/core";
import { environment } from "../../../environments/environment";
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { RolDTO, UserDTO, UserRequestDTO, UserResponseDTO } from "../models/user.models";





@Injectable({
  providedIn: 'root',
})


export class UserService {

  private readonly baseUrl = environment.apiUrl;

  constructor(private http: HttpClient){}

  me(): Observable<UserDTO>{

    return this.http.get<UserDTO>(`${this.baseUrl}/api/v1/user/me`);
  }

  updateUser(payload: UserRequestDTO): Observable<UserResponseDTO>{
    return this.http.put<UserResponseDTO>(`${this.baseUrl}/api/v1/user`,payload);
  }



}

File: backend/src/main/java/com/demos/service/UserService.java
package com.demos.service;

import com.demos.model.dto.UserDTO;
import com.demos.model.dto.UserRequestDTO;
import com.demos.model.dto.UserResponseDTO;

public interface UserService {

	public UserResponseDTO updateUser(UserRequestDTO userRequestDTO);
	public UserDTO getUser(Long id);
	public UserDTO getUserByEmail(String email);
	
}
